### PR TITLE
Websocketpp implementation for win32 + mobile

### PR DESCRIPTION
**Pull request by jplafonta**

### CODE REVIEW

File: Source/WebSocket/Win32/win32_websocket.cpp [hunk 1]
```diff
@@ -3,9 +3,601 @@
...
+        }
+
+        // At this point, there should be no more references to me.
+        m_state = DESTROYED;
+    }
+
+    //std::shared_ptr<wspp_websocket_impl> shared_from_this()
+    //{
+    //    return std::static_pointer_cast<wspp_websocket_impl>(hc_task::shared_from_this());
+    //}
```
Comment (jplafonta) [inline Source/WebSocket/Win32/win32_websocket.cpp]: remove this

File: Source/WebSocket/Win32/win32_websocket.cpp [hunk 2]
```diff
@@ -3,9 +3,601 @@
...
+    {
+        m_hcWebsocketHandle = HCWebSocketDuplicateHandle(hcHandle);
+    }
+
+    ~wspp_websocket_impl()
+    {
+        _ASSERTE(m_state < DESTROYED);
+
+        // Now, what states could we be in?
+        switch (m_state) {
```
Comment (jasonsandlin) [inline Source/WebSocket/Win32/win32_websocket.cpp]: >) { [](start = 23, length = 3)

nit: new line {

File: Source/WebSocket/Win32/win32_websocket.cpp [hunk 3]
```diff
@@ -3,9 +3,601 @@
...
+    {
+        auto &client = m_client->client<WebsocketConfigType>();
+        client.stop_perpetual();
+
+        // Can't join thread directly since it is the current thread.
+        AsyncBlock* async = new (xbox::httpclient::http_memory::mem_alloc(sizeof(AsyncBlock))) AsyncBlock {};
+        async->callback = [](AsyncBlock* async)
+        {
+            xbox::httpclient::http_memory::mem_free(async);
+        };
```
Comment (jasonsandlin) [inline Source/WebSocket/Win32/win32_websocket.cpp]: set queue

File: Source/WebSocket/Win32/win32_websocket.cpp [hunk 4]
```diff
@@ -3,9 +3,601 @@
...
+        virtual websocketpp::client<websocketpp::config::asio_client> & non_tls_client()
+        {
+            throw std::bad_cast();
+        }
+        virtual websocketpp::client<websocketpp::config::asio_tls_client> & tls_client()
+        {
+            throw std::bad_cast();
+        }
+        virtual bool is_tls_client() const = 0;
+    };
```
Comment (jasonsandlin) [inline Source/WebSocket/Win32/win32_websocket.cpp]: nit: newline after

File: Source/WebSocket/Win32/win32_websocket.cpp [hunk 5]
```diff
@@ -3,9 +3,601 @@
...
+    struct websocketpp_client : websocketpp_client_base
+    {
+        websocketpp::client<websocketpp::config::asio_client> & non_tls_client() override
+        {
+            return m_client;
+        }
+        bool is_tls_client() const override { return false; }
+        websocketpp::client<websocketpp::config::asio_client> m_client;
+    };
+    struct websocketpp_tls_client : websocketpp_client_base
```
Comment (jasonsandlin) [inline Source/WebSocket/Win32/win32_websocket.cpp]: nit: newline after

File: Source/WebSocket/Win32/win32_websocket.cpp [hunk 6]
```diff
@@ -3,9 +3,601 @@
...
+    // only happen once for each client.
+#if defined(__APPLE__) || (defined(ANDROID) || defined(__ANDROID__)) || defined(_WIN32)
+    bool m_opensslFailed;
+#endif
+
+    hc_websocket_handle_t m_hcWebsocketHandle;
+
+    Uri m_uri;
+
+    const static http_internal_string m_subProtocolHeader;
```
Comment (jasonsandlin) [inline Source/WebSocket/Win32/win32_websocket.cpp]: don't make static internal strings or it'll break cause static is created before mem hook is set and then its dtor'd using the incorrect free.

File: Source/WebSocket/Win32/win32_websocket.cpp [hunk 7]
```diff
@@ -3,9 +3,601 @@
 
 #include "pch.h"
```
Comment (jasonsandlin) [inline Source/WebSocket/Win32/win32_websocket.cpp]: should name this websocketpp_websocket.cpp cause this works for mobile & win32

**Pull request merged**